netcreateorg/netcreate-itest
# Build Configuration Fixes

Ben reported a mystery error of `node_modules` being missing from the repo, and as these directories shouldn't be in the directory in the first place an investigation was spawned! The following changes were made:

1. removed files that had been committed by accident in `_ur/node_modules`
2. clearer the hidden files settings in the code-workspace, as `node_modules` was one of them which hid the accidental commit for a long time!
3. fixed bad `.gitignore` pattern `**/node_modules/*` to `**/node_modules` 
4. extra: fixed broken `npm run build-libs` scripts
5. extra: added `*_nocommit` to gitignore so temporary directories could be kept in the repo folder 

## TO TEST

* make sure you are running node `v18.18.2` (`node -v`) 
* make sure you are on `i386` architecture (`arch`)

next, in the main `netcreate-itest` directory:
* `npm run clean`
* `npm ci`
* `npm run dev`

next, check that the `node_modules` directories in the following directories are **grayed-out** and are not showing up as "modified files" in your git client.

* `_ur/node_modules`
* `_ur_mods/node_modules`
* `_node_modules`
